examples/evm-multichain-dapp/pages/index.tsx

```diff
@@ -0,0 +1,183 @@
+import { useWeb3React } from '@web3-react/core'
+import type { NextPage } from 'next'
+import Head from 'next/head'
+import Image from 'next/image'
+import { useEffect, useState } from 'react'
+import blockchains from '../config/blockchains.json'
+import { ChainId } from '../config/chainIds'
+import { NETWORK_ICON, NETWORK_LABEL } from '../config/networks'
+import { injected, supportedChainIds, wallets } from '../config/wallets'
+import useWeb3 from '../hooks/useWeb3'
+
+import styles from '../styles/Home.module.css'
+
+const Home: NextPage = () => {
+  const { account, activate, chainId, deactivate } = useWeb3React()
+  const [nativeBalance, setNativeBalance] = useState(0)
+  const [sendAddress, setSendAddress] = useState('')
+  const [sendAmount, setSendAmount] = useState('0')
+
+  const { web3 } = useWeb3()
+  // @ts-ignore TYPE NEEDS FIXING
+  const chainData = blockchains[chainId]
```

Comment:
Looks like we are pulling in a JSON file with ~9000 lines here. Is this really optimal? :thinking:

Maybe we could use `getStaticProps` and a [dynamic route](https://nextjs.org/docs/routing/dynamic-routes) with `fallback: false`.

Eg.:

`/blockchains/1`

Reply:
- We can reduce the file size to ensure data is available for only listed blockchains `config/chainIds.ts`.
- Agree, the dynamic route & getStaticProps can be implemented to optimize it according to chainId.